core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/inventory/service/InventoryServiceImpl.java

```diff
@@ -195,25 +204,31 @@ public void decrementInventory(Map<Sku, Integer> skuQuantities, Map<String, Obje
     protected void decrementSku(Map<Sku, Integer> skuQuantities, Map<String, Object> context) throws InventoryUnavailableException {
         for (Entry<Sku, Integer> entry : skuQuantities.entrySet()) {
             Sku sku = entry.getKey();
+            Sku skuForInventory = sku;
+            if (sku.getProduct().getUseDefaultSkuInInventory()) {
+                skuForInventory = sku.getProduct().getDefaultSku();
+            }
             Integer quantity = entry.getValue();
             if (quantity == null || quantity < 1) {
                 throw new IllegalArgumentException("Quantity " + quantity + " is not valid. Must be greater than zero and not null.");
             }
 
-            if (checkBasicAvailablility(sku)) {
-                if (InventoryType.CHECK_QUANTITY.equals(sku.getInventoryType())) {
-                    Integer inventoryAvailable = retrieveQuantityAvailable(sku, context);
+            if (checkBasicAvailablility(skuForInventory)) {
+                if (InventoryType.CHECK_QUANTITY.equals(skuForInventory.getInventoryType())) {
+                    Integer inventoryAvailable = retrieveQuantityAvailable(skuForInventory, context);
                     if (inventoryAvailable == null) {
                         return;
                     }
                     if (inventoryAvailable < quantity) {
                         throw new InventoryUnavailableException(
-                                "There was not enough inventory to fulfill this request.", sku.getId(), quantity, inventoryAvailable);
+                                "There was not enough inventory to fulfill this request.", skuForInventory.getId(), quantity, inventoryAvailable);
                     }
                     int newInventory = inventoryAvailable - quantity;
-                    sku.setQuantityAvailable(newInventory);
+                    skuForInventory.setQuantityAvailable(newInventory);
                     catalogService.saveSku(sku);
+                    catalogService.saveSku(skuForInventory);
```

Comment:
Both sku and skuForInventory need to be saved/invalidated?

Reply:
@danielcolgrove Does this need more changes?  Could you review it again?